modules/extensions-support/src/main/java/org/mule/runtime/module/extension/api/metadata/PropagatedParameterTypeResolver.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2023 Salesforce, Inc. All rights reserved.
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.api.metadata;
+
+import static java.util.Optional.empty;
+
+import org.mule.metadata.api.model.MetadataType;
+import org.mule.runtime.api.metadata.MetadataResolvingException;
+
+import java.util.Optional;
+
+/**
+ * Resolves the propagated parameter type by name.
+ *
+ * @since 4.8
```

Comment:
shouldn't it be 4.9?

Reply:
we are going to cherry pick this, so I was already declaring it as such